core/src/main/java/org/pentaho/di/connections/ConnectionManager.java

```diff
@@ -291,6 +292,20 @@ public <T extends ConnectionDetails> boolean test( T connectionDetails ) throws
     return connectionProvider.test( connectionDetails );
   }
 
+  /**
+   * Run a test operation on the named connection
+   *
+   * @param connectionDetails The named connection details to test
+   * @return A boolean signifying the success of the test operation
+   */
+  @SuppressWarnings( "unchecked" )
+  public <T extends VFSConnectionDetails> boolean test(T connectionDetails, VFSConnectionTestOptions options )
+          throws KettleException {
+    VFSConnectionProvider<T> connectionProvider =
+            (VFSConnectionProvider<T>) connectionProviders.get( connectionDetails.getType() );
+    return connectionProvider.test( connectionDetails, options );
```

Comment:
Another point of discussion, How/where is `#test(ConnectionDetails, VFSConnectionTestOptions)` being called?

Currently PDI File Open Browser has test functionality and it's calling

`result = connectionManager.test( connectionDetails );`
https://github.com/pentaho/pentaho-kettle/blob/7bb5b14e652642ed613e49db3fdfece1186088c1/plugins/connections/ui/src/main/java/org/pentaho/di/vfs/connections/ui/dialog/ConnectionDialog.java#L408-L414

Then the code does to ConnectionManager#test

https://github.com/pentaho/pentaho-kettle/blob/7bb5b14e652642ed613e49db3fdfece1186088c1/core/src/main/java/org/pentaho/di/connections/ConnectionManager.java#L288-L292


I wouldn't want to have to cast down to BaseVFSConnectionProvider to execute the rootPath test. 

Shouldn't this be more of the approach

pseudo code for ConnectionManager
```
@Override
public boolean test( T ConnectionDetails) {
  if(somecondition) { // decide if to use legacy test
    super.test(connectionDetails);
  }
  else {
    VFSConnectionTestOptions testOptions =  // create options based on ConnectionDetails
    test(connectionDetails,  testOptions);
  }

}
```

Reply:
The #test(ConnectionDetails, VFSConnectionTestOptions) is being called in ConnectionManagerService. See PR - https://github.com/pentaho/pentaho-scheduler-plugin-ee/pull/223. 

We are passing ignoreRootPath as query param from ConnectionManagerResource which in turn would be calling #test(ConnectionDetails, VFSConnectionTestOptions) . Hence the method is kept isolated from #test(ConnectionDetails).

Reply:
You can go either way. Not a major issue to prevent a PR merge. But are you passing parameters that would reside in the ConnectionDetails object? If you are shouldn't the values from connectionDetails determine if you do `test(ConnectionDetails) `or `test(ConnectionDetails, options)`.